src/shared/components/comment/comment-report.tsx

```diff
@@ -172,4 +231,38 @@ export class CommentReport extends Component<
       resolved: !i.props.report.comment_report.resolved,
     });
   }
+
+  handleRemoveComment() {
+    this.setState({
+      loading: true,
+      removeCommentForm: {
+        comment_id: this.props.report.comment_report.comment_id,
+        removed: !this.props.report.comment.removed,
+      },
+    });
+  }
+
+  async handleRemoveComment2(reason: string) {
+    var form = this.state.removeCommentForm!;
+    form.reason = reason;
+    this.props.onRemoveComment(form);
+    this.setState({ removeCommentForm: undefined });
+  }
+
+  handleModBanFromCommunity() {
+    this.setState({ loading: true });
+    this.props.onModBanFromCommunity({
+      person: this.props.report.comment_creator,
+      community: this.props.report.community,
+      ban: !this.props.report.creator_banned,
+    });
```

Comment:
This is the weird thing about not having these modals inside here. Now its pushing up half a form, without the reason included.